zoom_rooms/indico_zoom_rooms/plugin.py

```diff
@@ -0,0 +1,271 @@
+    assoc: VCRoomEventAssociation,
+    vc_room: VCRoom,
+    old_link: Event | Contribution | SessionBlock,
+    event: Event,
+    data: dict | None = None,
+):
+    if vc_room.type == 'zoom':
+        # when a room is detached from an event/contribution/session block, delete the corresponding zoom room entry
+        _delete_zoom_association(old_link, vc_room)
+
+
+def _signal_zoom_meeting_data_updated(vc_room: VCRoom, data: dict):
+    if vc_room.type == 'zoom' and (name := data.get('name')) and name != vc_room.name:
+        for assoc in vc_room.events:
+            if zr_id := get_zoom_room_id(assoc.link_object):
+                ZoomRoomsQueueEntry.record(ZoomRoomsAction.update, zr_id, assoc=assoc, args=OperationArgs(title=name))
+
+
+class ZoomRoomsPlugin(IndicoPlugin):
+    """Zoom Rooms
+
+    Zoom Rooms / Exchange synchronization plugin for Indico."""
+
+    configurable = True
+    settings_form = PluginSettingsForm
+    default_settings = {'debug': False, 'service_url': None, 'docs_url': '', 'token': None, 'timeout': 3}
+    category = PluginCategory.videoconference
+
+    def init(self):
+        super().init()
+        # Here we plug the various object signals into the action tracking code
+        # Consult the README.md for a summary of the logic behind
+        self.connect(signals.event.contribution_updated, _signal_link_object_updated)
+        self.connect(signals.event.session_block_updated, _signal_link_object_updated)
+        self.connect(signals.event.updated, _signal_link_object_updated)
+        self.connect(signals.event.times_changed, _signal_tt_entry_updated, sender=Contribution)
+        self.connect(signals.event.times_changed, _signal_tt_entry_updated, sender=SessionBlock)
+        self.connect(signals.vc.new_vc_room, _signal_zoom_meeting_created)
+        self.connect(signals.vc.cloned_vc_room, _signal_zoom_meeting_cloned)
+        self.connect(signals.vc.attached_vc_room, _signal_zoom_meeting_association_attached)
+        self.connect(signals.vc.detached_vc_room, _signal_zoom_meeting_association_detached)
+        self.connect(signals.vc.updated_vc_room_data, _signal_zoom_meeting_data_updated)
+
+        self.template_hook('manage-event-vc-extra-buttons', self.inject_button)
+        self.template_hook('event-vc-extra-buttons', self.inject_button)
+        self.template_hook('event-timetable-vc-extra-buttons', self.inject_button)
+
+    def inject_button(self, event_vc_room: VCRoomEventAssociation, **kwargs: dict):
+        if event_vc_room.vc_room.type != 'zoom' or not event_vc_room.link_object.room:
+            return
+
+        if zr_id := get_zoom_room_id(event_vc_room.link_object):
+            return render_plugin_template(
+                'explanation.html',
+                room_name=event_vc_room.link_object.room.full_name,
+                event_vc_room=event_vc_room,
+                zr_id=zr_id,
+                docs_url=ZoomRoomsPlugin.settings.get('docs_url'),
+                **kwargs,
```

Comment:
I would not pass along random kwargs - the signal handler needs to accept them to be forwards-compatible in case we start passing anything new, but since your template does not expect anything besides the `event_vc_room`, I don't think it's a good idea to pass extra ones.